x-pack/plugins/triggers_actions_ui/public/application/sections/rules_list/components/notify_badge/notify_badge.tsx

```diff
@@ -30,35 +30,38 @@ import {
 } from './translations';
 import { RulesListNotifyBadgeProps } from './types';
 
-export const RulesListNotifyBadge: React.FunctionComponent<RulesListNotifyBadgeProps> = (props) => {
-  const {
-    isLoading = false,
-    rule,
-    isOpen,
-    onClick,
-    onClose,
-    onLoading,
-    onRuleChanged,
-    snoozeRule,
-    unsnoozeRule,
-    showOnHover = false,
-    showTooltipInline = false,
-  } = props;
-
-  const { isSnoozedUntil, muteAll, isEditable } = rule;
-
+export const RulesListNotifyBadge: React.FunctionComponent<RulesListNotifyBadgeProps> = ({
+  snoozeSettings,
+  loading = false,
+  disabled = false,
+  onRuleChanged,
+  snoozeRule,
+  unsnoozeRule,
+  showOnHover = false,
+  showTooltipInline = false,
+}) => {
+  const [requestInFlight, setRequestInFlightLoading] = useState(false);
+  const isLoading = loading || requestInFlight;
+  const isDisabled = !!disabled || !snoozeSettings;
```

Comment:
The only reason for `!!disabled` here is it has a type `boolean | string` where `string` means the component is disabled and gives a human readable reason. It's necessary when a user doesn't have credentials to change rule snooze settings or it wasn't possible to fetch the data. Another option is to have `disabledReason: string` prop and it could be more readable while it makes reasoning about what the component shows when `disabled = false` and `disabledReason = 'some string'` difficult and requires checking the docs or the implementation. While having one `disabled` prop is rather straightforward.

It's a bummer there is more than one way to do the same thing, `!!` is just much shorter than `Boolean()` but I can change to `Boolean()` as you prefer.